archtec/scripts/playtime.lua

```diff
@@ -17,45 +13,31 @@ end
 function archtec.get_total_playtime(name)
 	local player = minetest.get_player_by_name(name)
 	if player then
-		return player:get_meta():get_int(key) + archtec.get_session_playtime(name)
+		return archtec_playerdata.get(name, "playtime") + archtec.get_session_playtime(name)
 	end
 end
 
+function archtec.playtimesave(name)
+	archtec_playerdata.mod(name, "playtime", archtec.get_session_playtime(name))
```

Comment:
Since `playtimesave` gets called periodically - looks  like once every 20s from `archtech_playerdata/init.lua` - this will over-increment the time, constantly. In about 20mins of playing, I "gained" about 4 hours of playtime.

One solution could be to immediately reset `current[name] = os.time()` after the `mod(..)` function call. That fixes it, but will reset the session time to 0 - even though session time is not currently a stats being shown, you may want to show it at some point.

The other solution is to have another table like `playtime_before_current` - then on join do (and reset to nil on leave):
```lua
playtime_before_current[name] = archtec_playerdata.get(name, "playtime")
```
and then modify the functions as appropreate:
```lua
function archtec.get_total_playtime(name)
  local player = minetest.get_player_by_name(name)
  if player then
    return playtime_before_current[name] + archtec.get_session_playtime(name)
  end
end

function archtec.playtimesave(name)
  archtec_playerdata.set(name, "playtime", playtime_before_current[name] + archtec.get_session_playtime(name))
end
```